flang/lib/Lower/CallInterface.cpp

```diff
@@ -31,7 +31,11 @@ bool Fortran::lower::CallerInterface::hasAlternateReturns() const {
 std::string Fortran::lower::CallerInterface::getMangledName() const {
   const auto &proc = procRef.proc();
   if (const auto *symbol = proc.GetSymbol())
-    return converter.mangleName(*symbol);
+    // Do NOT mangle names for functions/subroutines with BIND(C) attribute.
+    // These could be potential calls to external RTL (i.e OpenMP), that may not
+    // follow the same mangling semantics.
+    return IsBindCProcedure(*symbol) ? proc.GetName()
```

Comment:
Thanks @SouraVX for fixing this.

Will this return "print_c" or "print_C" now for the following case?
```
    subroutine print_c(string) bind(C, name="print_C")
```

Reply:
I think it will return "print_c" which is not what we want here. The bind name, if there is one, can be found in the `semantics::SubprogramDetails`.
Since you are on the callee side here, I think you have to be careful since `symbol` may be a have `ProcEntityDetails`, in which case you should look for a potential bind name in its interface symbol (see [IsBindCProcedure impl](SubprogramDetails) for an example of how to navigate up to the interface symbol).
What you'll get from `SubprogramDetails::bindName()`, if anything, is an expression, but it must be a scalar constant as per Fortran R808 grammar rule, so you can unwrap a constant from that and get the string.